### ISSUE TITLE
Add optional WCAG 2.1 support

### ISSUE THREAD

**Issue by mgifford** [GSA/openacr#270]
I've heard from a few accessibility consultants now that because OpenACR is still using WCAG 2.0 that they are less likely to start testing it with some of their projects.  WCAG 2.1 is the current best practice for the private sector. WCAG 2.2 probably will be next year.  

The Whitehouse and other agencies are currently aiming for WCAG 2.1 guidelines because they are the current release candidate for 2022. Supporting this will aid with adoption, even if WCAG 2.0 is all that is technically required for Section 508.

**Comment by dmundra** [comment]
This would be good to bring up with Andrew and Syed for prioritization.

**Comment by drewnielson** [comment]
Will defer to Syed, but I agree that supporting WCAG 2.1 and 2.2 would make OpenACR much more appealing, including to US govt. vendors who also sell products internationally. By providing them with a single option to create their ACRs, we would still be helping US agencies more easily obtain and share ACR data.

**Comment by saz33m** [comment]
WCAG 2.1 makes sense, but what's the trade off here?

**Comment by dmundra** [comment]
From a technical perspective, supporting WCAG 2.1 as a separately generated OpenACR (basing it on VPAT 2.4 WCAG version) would be the first time we would be tackling supporting another VPAT format (which we did envision that will happen at some point). This would require modifying both the validator and editor to allow the user to toggle/switch supported versions. The trade-off here is that we would be adding a lot of new functionality with a slightly different user experience before refining/adjusting what we have now. If we are ready to that then we can work on this next.

@mgifford are there other items to consider?

**Comment by mitchellevan** [comment]
> I've heard from a few accessibility consultants now that because OpenACR
is still using WCAG 2.0 that they are less likely to start testing it with
some of their projects. WCAG 2.1 is the current best practice for the
private sector.

TPGi is one such consultant. The large majority of our ACRs for clients
include WCAG 2.1.

> The trade-off here is that we would be adding a lot of new functionality
with a slightly different user experience before refining/adjusting what we
have now.

The online oACR editor is essential so anybody can create one, but it
doesn't fit our workflow at the scale of hundreds or thousands of ACRs. We
would need to agree on the YAML format to support WCAG 2.1 so we can
generate it from our own tools, not through the online editor.

Another possibility is that US Fed would require a 508-only version of
their YAML file, but this would be a new requirement on sellers. Today
agencies must already be getting lots of ACRs with 508+x because that's the
ITIC way with the VPAT "editions."

I propose you prioritize support for plus-standards in the YAML, starting
with WCAG 2.1, even if the online editor doesn't support 2.1 in the near
term.

If the status quo is burdensome to agencies, then give the agencies a way
to filter out the plus-standards when they view and compare ACRs.

**Comment by mgifford** [comment]
Thanks for this @mitchellevan for industrial scale ACRs like you are doing, worth checking out the command line documentation here:
https://github.com/GSA/openacr/tree/main/docs


We've built this anticipating that we need to support WCAG 2.x & EN 301 549 requirements eventually, so we built a Librarian to help compile the relevant rules for the schema.  

Once the command line tools work, then we'll work on the UI side with the editor. Ultimately, I would like to see folks build for good customer experience, rather than to guidelines that were a best practice in 2008.

**Comment by bruce-usab** [comment]
I will take the liberty to note that [2.2 is expected to achieve TR status in June](https://www.w3.org/WAI/standards-guidelines/wcag/new-in-22/#introduction-timeline-comments).

I will also acknowledge the challenge for oACR to provide good usability for a diverse range of users.  Some regard AAA and/or 2.1/2.2 SC as noise.  Some (e.g., CA/EU) include 2.1 SC in the minimal baseline.  And some users will want oACR to incorporate 2.2 as soon as is technically feasible, regardless of force of law.

**Comment by dmundra** [comment]
Started work on this https://github.com/GSA/openacr/tree/wcag-2.1-support. Reviewing the WCAG 2.1 VPAT I found one interesting thing:

WCAG 2.1 VPAT chapters for A,AA,AAA link 2.0 criteria to https://www.w3.org/TR/WCAG20/ and 2.1 criteria to https://www.w3.org/TR/WCAG21/ instead of linking all criteria to the 2.1 URL. 2.0 items are listed under the 2.1 URL just with different anchor links. Wonder why the want to direct users to different URLs.

I can recreate that expectation in the report but I have to add data to distinguish which URL to use for the criteria link.

**Comment by mgifford** [comment]
2.1 includes 2.0. I suspect that The WCAG 2.1 VPAT links to the older 2.0 simply because they were just Word Documents, and it was easier to just leave them pointing to 2.0, rather than adjust to 2.1.  I would align with pointing to the WCAG criteria that isn't a decade old if we can.

**Comment by mgifford** [comment]
For folks who want to use WCAG 2.0 OpenACR, I don't see any concern with pointing them to a description for 2.1, but think it would be better if it pointed to the 2008 WCAG 2.0 documents.  Unfortunately...  

If we've got them all bundled together, let's just point all the the links to WCAG 2.1.

**Comment by mitchellevan** [comment]
I agree, point all links to WCAG 2.1.

A disadvantage is it might look like an error to someone focused solely on Section 508, which includes only WCAG 2.0 by reference. But as Mike correctly said, the normative text of the guidelines is identical, and the informative text within 2.1 is better (Understanding, Techniques).

VPAT also includes a link to WCAG Conformance Requirements, where 2.2 added a third Note under 5.2.2 Full Pages.

**Comment by dmundra** [comment]
Thanks @mgifford and @mitchellevan. I created a specific 2.1 catalog and also 2.1+508 catalog. I run with this and see if we can get a working solution that allows you to select different catalogs.

**Comment by dmundra** [comment]
Working on the editor https://github.com/GSA/openacr-editor/tree/wcag-2.1-multiple-catalogs. Not ready for a pull request yet.

I was able to get the editor to switch different catalogs but I had to handle when switching from 2.0 to 2.1 adding the 2.1 criteria and if you switch from 2.1 back to 2.0 removing the new criteria. Now this approach is destructive of data but if we leave 2.1 criteria in the OpenACR and switch to 2.0 the validator will complain about invalid criteria. Which I believe is what we want.

Here is what the catalog choice looks like on the overview page (I decided to add it there as the first thing to do):

![Screenshot of catalog choice](https://user-images.githubusercontent.com/7096681/155433028-8b3ea84d-5bdd-47d2-bd5d-3dae1a270ede.png)
